lib/matplotlib/patches.py

```diff
@@ -2140,6 +2115,44 @@ def segment_circle_intersect(x0, y0, x1, y1):
         # restore original path
         self._path = path_original
 
+    def _update_path(self):
+        # Compute new values and update and set new _path if changed
+        stretched = self._theta_stretch()
+        if any(a != b for a, b in zip(stretched, (self._theta1, self._theta2,
```

Comment:
Why this conditional?

Reply:
To not create a new path if not required. `Path.arc` is if nothing else more complicated than this test.